pytensor/tensor/rewriting/linalg.py

```diff
@@ -966,3 +970,22 @@ def rewrite_cholesky_diag_to_sqrt_diag(fgraph, node):
             non_eye_input = pt.shape_padaxis(non_eye_input, -2)
 
     return [eye_input * (non_eye_input**0.5)]
+
+
+@node_rewriter([_solve_bilinear_discrete_lyapunov])
```

Comment:
Instead of a rewrite make this the JAX dispatch of solve_bilinear_discrete_lyapunov (that is write the JAX code that's equivalent to the direct method as the dispatch)?

The problem is you're introducing this form rather late, after rewrites that would have helped JAX make sense of this graph.

Reply:
Couldn't I just register the rewriter sooner then? It's already at 0.9, so it happens before canonicalization

Reply:
Yes, ignore me muchacho